File: src/main/java/com/isfin/islamicfinancial/Entities/Company.java
package com.isfin.islamicfinancial.Entities;

import jakarta.persistence.*;
import java.util.List;

@Entity
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String sector;
    private String ticker;
    private double marketCap;

    @OneToOne(mappedBy = "company", cascade = CascadeType.ALL)
    private FinancialRatios financialRatios;

    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL)
    private List<ShariahCompilance> complianceRecords;

    // Getters and Setters
}

File: mobile/lib/models/company.dart
class Company {
  final int id;
  final String name;
  final String sector;
  final String ticker;
  final double marketCap;

  Company({
    required this.id,
    required this.name,
    required this.sector,
    required this.ticker,
    required this.marketCap,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['id'] as int,
      name: json['name'] ?? '',
      sector: json['sector'] ?? '',
      ticker: json['ticker'] ?? '',
      marketCap: (json['marketCap'] is int)
          ? (json['marketCap'] as int).toDouble()
          : (json['marketCap'] as double? ?? 0.0),
    );
  }
}
